big_int.cpp

```diff
@@ -0,0 +1,519 @@
+
+#include <bits/stdc++.h>
+#define ll long long
+using namespace std;
+
+class BigInteger
+{
+public:
+string str;
+int sign;
+
+  BigInteger()
+  {
+    sign=0;
+    str="0";
+  }
+  BigInteger(string s)
+  { 
+    if(s[0]=='-')
+     { sign=1;
+      str=s.substr(1);}
+      else{
+    sign=0;str=s;}
+  }
+  
+  BigInteger(ll s)
+  { sign=0;
+    if(s<0)	
+     { sign=1;}
+   str=to_string(s);
+  }
+
+friend bool operator< (BigInteger  , BigInteger  );
+friend bool operator> (BigInteger  , BigInteger  );
+friend bool operator<= (BigInteger  , BigInteger  );
+friend bool operator>= (BigInteger  , BigInteger  );
+friend bool operator== (BigInteger  , BigInteger  );
+friend bool operator!= (BigInteger  , BigInteger  );
+friend BigInteger operator+ (BigInteger  , BigInteger  );
+friend BigInteger operator- (BigInteger  , BigInteger  );
+friend BigInteger operator* (BigInteger  , BigInteger  );
+friend BigInteger operator/ (BigInteger  , BigInteger  );
+friend BigInteger operator% (BigInteger  , BigInteger  );
+friend ostream & operator<< (ostream &out, const BigInteger &a);
+
+
+
+
+
+};
+
+bool operator< (BigInteger  a, BigInteger  b) {
+  string str1=a.str;
+  string str2=b.str;
+  if (a.sign != b.sign){return a.sign>b.sign;}
+    else if(str1.size()!=str2.size()){//cout<<(a.size>b.size);
+    return str1.size()<str2.size();}
+  else
+  { 
+    for(int i=0;i<str1.size();i++)
+    {
+           if(str1[i] != str2[i])
+            {//cout<<(temp->data > temp1->data);
+              return (str1[i]-'0')< (str2[i]-'0');}
+              }
+  }
+  return false;
+  }
+
+  
+  bool operator== (BigInteger  a, BigInteger  b) {
+     string str1=a.str;
+  string str2=b.str;
+  if (a.sign != b.sign){return false;}
+  else if(str1.size()!=str2.size()){
+    return false;}
+  else
+  { 
+    for(int i=0;i<str1.size();i++)
+    {
+           if(str1[i] != str2[i])
+            {
+              return false;}
+           
+    }
+  }
+  return true;
+  }
+  bool operator!= (BigInteger  a, BigInteger  b) {
+  return !(a==b);
+  }
+  bool operator<= (BigInteger  a, BigInteger  b) {
+  return (a<b || a==b);
+  }
+  bool operator> (BigInteger  a, BigInteger  b) {
```

Comment:
Codacy found an issue: [Function parameter 'a' should be passed by const reference.](https://app.codacy.com/gh/ayaankhan98/bigINT/pullRequest?prid=6290647)